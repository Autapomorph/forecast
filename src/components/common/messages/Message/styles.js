import styled from 'styled-components';

export const StyledMessageWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  padding: 10px;
  border-radius: 5px;
`;

export const StyledMessageContent = styled.span`
  text-align: center;
  font-size: 1.5rem;
`;
