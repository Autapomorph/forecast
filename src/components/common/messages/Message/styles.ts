import styled from 'styled-components/macro';

export const StyledMessageWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  padding: 10px;
  border-radius: 5px;
  font-size: 1.3rem;
`;

export const StyledMessageContent = styled.span`
  text-align: center;
`;
