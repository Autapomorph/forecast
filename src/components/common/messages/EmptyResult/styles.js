import styled from 'styled-components';

import Message from '../Message';

// eslint-disable-next-line import/prefer-default-export
export const StyledEmptyListMessage = styled(Message)``;
