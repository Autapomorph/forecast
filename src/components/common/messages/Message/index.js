import React from 'react';

import { StyledMessageWrapper, StyledMessageContent } from './styles';

const Message = ({ children, ...props }) => (
  <StyledMessageWrapper {...props}>
    <StyledMessageContent>{children}</StyledMessageContent>
  </StyledMessageWrapper>
);

export default Message;
