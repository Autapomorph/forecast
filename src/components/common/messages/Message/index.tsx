import React from 'react';

import { StyledMessageWrapper, StyledMessageContent } from './styles';

const Message: React.FC = ({ children, ...props }): React.ReactElement => (
  <StyledMessageWrapper {...props}>
    <StyledMessageContent>{children}</StyledMessageContent>
  </StyledMessageWrapper>
);

export default Message;
