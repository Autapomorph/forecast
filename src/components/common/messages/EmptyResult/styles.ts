import styled from 'styled-components/macro';

import Message from 'components/common/messages/Message';

export const EmptyResultMessage = styled(Message)`
  margin: 20px 0;
`;
