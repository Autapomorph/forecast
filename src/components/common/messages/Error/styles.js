import styled from 'styled-components';

import Message from '../Message';

// eslint-disable-next-line import/prefer-default-export
export const StyledErrorMessage = styled(Message)`
  background: var(--bg-error);
  color: var(--error-text-color);
  font-size: 1.3rem;
`;
