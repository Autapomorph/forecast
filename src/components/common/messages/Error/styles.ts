import styled from 'styled-components/macro';

import Message from 'components/common/messages/Message';

export const StyledErrorMessage = styled(Message)`
  background: var(--bg-error);
  color: var(--error-text-color);
`;
