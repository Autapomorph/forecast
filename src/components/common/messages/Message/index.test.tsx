import { shallow } from 'enzyme';

import Message from '.';

import { StyledMessageContent } from './styles';

describe('Message tests', () => {
  it('renders properly', () => {
    const wrapper = shallow(<Message />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render children', () => {
    const text = 'text';
    const wrapper = shallow(<Message>{text}</Message>);

    expect(wrapper.find(StyledMessageContent).text()).toEqual(text);
  });
});
