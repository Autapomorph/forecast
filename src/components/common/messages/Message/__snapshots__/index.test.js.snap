// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message tests renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Message />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
  text-align: center;
  font-size: 1.5rem;
",
          ],
        },
        "displayName": "styled.span",
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "span",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  padding: 10px;
  border-radius: 5px;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
  text-align: center;
  font-size: 1.5rem;
",
            ],
          },
          "displayName": "styled.span",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "span",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  padding: 10px;
  border-radius: 5px;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
